src/main/java/com/pinterest/secor/io/impl/FlexibleDelimitedFileReaderWriterFactory.java

```diff
@@ -0,0 +1,166 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.pinterest.secor.io.impl;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+import com.pinterest.secor.io.FileReader;
+import com.pinterest.secor.io.FileReaderWriterFactory;
+import com.pinterest.secor.io.FileWriter;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.Compressor;
+import org.apache.hadoop.io.compress.CodecPool;
+import org.apache.hadoop.io.compress.Decompressor;
+
+import org.apache.commons.configuration.ConfigurationException;
+
+import com.google.common.io.CountingOutputStream;
+import com.pinterest.secor.common.LogFilePath;
+import com.pinterest.secor.io.KeyValue;
+import com.pinterest.secor.util.FileUtil;
+import com.pinterest.secor.common.SecorConfig;
+
+/**
+* Flexible Delimited Text File Reader Writer with Compression
+*
+* @author Ahsan Nabi Dar (ahsan@wego.com)
+*/
+public class FlexibleDelimitedFileReaderWriterFactory implements FileReaderWriterFactory {
+
+  @Override
+  public FileReader BuildFileReader(LogFilePath logFilePath, CompressionCodec codec)
+  throws IllegalAccessException, IOException, InstantiationException {
+    return new FlexibleDelimitedFileReader(logFilePath, codec);
+  }
+
+  @Override
+  public FileWriter BuildFileWriter(LogFilePath logFilePath, CompressionCodec codec) throws IOException {
+    return new FlexibleDelimitedFileWriter(logFilePath, codec);
+  }
+
+
+
+  protected class FlexibleDelimitedFileReader implements FileReader {
+    private final BufferedInputStream mReader;
+    private long mOffset;
+    private Decompressor mDecompressor = null;
+    private byte DELIMITER = delimterReader();
+
+    public FlexibleDelimitedFileReader(LogFilePath path, CompressionCodec codec) throws IOException {
+      Path fsPath = new Path(path.getLogFilePath());
+      FileSystem fs = FileUtil.getFileSystem(path.getLogFilePath());
+      InputStream inputStream = fs.open(fsPath);
+      this.mReader = (codec == null) ? new BufferedInputStream(inputStream)
+      : new BufferedInputStream(
+      codec.createInputStream(inputStream,
+      mDecompressor = CodecPool.getDecompressor(codec)));
+      this.mOffset = path.getOffset();
+    }
+
+    public byte delimterReader() {
+      byte delimiter = '\n';
+      try {
+        String readerDelimiter = SecorConfig.load().getFileReaderDelimiter();
+        if (!readerDelimiter.isEmpty()){
+          delimiter = (byte)readerDelimiter.charAt(0);
+        }
+      }catch(ConfigurationException e){}
```

Comment:
Coding style: need a space between } and catch
Usually you should not ignore the Exception, it's better to throw it up or convert to a RuntimeException
The method name is better named as getReaderDelimter()